test/OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth.test.cpp

```diff
@@ -259,3 +244,111 @@ TEST(OpenSpaceToolkit_Physics_Environment_Atmospheric_Earth, GetDensityAt_LLA)
         );
     }
 }
+
+TEST(OpenSpaceToolkit_Physics_Environment_Atmospheric_Earth, GetDensityAt_Frames)
+{
+    {
+        static const Array<Tuple<EarthAtmosphericModel::Type, LLA, Instant, Real>> testCases = {
+            {EarthAtmosphericModel::Type::Exponential,
+             LLA(Angle::Degrees(35.076832), Angle::Degrees(-92.546296), Length::Kilometers(123.0)),
+             Instant::J2000(),
+             1e-13},
+            {EarthAtmosphericModel::Type::Exponential,
+             LLA(Angle::Degrees(35.076832), Angle::Degrees(-92.546296), Length::Kilometers(499.0)),
+             Instant::J2000(),
+             1e-15},
+            {EarthAtmosphericModel::Type::Exponential,
+             LLA(Angle::Degrees(35.076832), Angle::Degrees(-92.546296), Length::Kilometers(501.0)),
+             Instant::J2000(),
+             1e-15},
+            {EarthAtmosphericModel::Type::Exponential,
+             LLA(Angle::Degrees(35.076832), Angle::Degrees(-92.546296), Length::Kilometers(550.0)),
+             Instant::J2000(),
+             1e-15},
+            {EarthAtmosphericModel::Type::NRLMSISE00,
+             LLA(Angle::Degrees(0.0), Angle::Degrees(0.0), Length::Kilometers(500.0)),
+             Instant::DateTime(DateTime::Parse("2021-01-01 00:00:00"), Scale::UTC),
+             1e-15}
+        };
+
+        for (const auto& testCase : testCases)
+        {
+            const LLA lla = std::get<1>(testCase);
+
+            const Position positionITRF = {
+                lla.toCartesian(
+                    EarthGravitationalModel::EGM2008.equatorialRadius_, EarthGravitationalModel::EGM2008.flattening_
+                ),
+                Position::Unit::Meter,
+                Frame::ITRF()
+            };
+
+            const EarthAtmosphericModel earthAtmosphericModelITRF = {std::get<0>(testCase), Frame::ITRF()};
+            const EarthAtmosphericModel earthAtmosphericModelTEME = {std::get<0>(testCase), Frame::TEME()};
+
+            const Instant instant = std::get<2>(testCase);
+            const Real tolerance = std::get<3>(testCase);
+
+            const Real densityITRF = earthAtmosphericModelITRF.getDensityAt(positionITRF, instant);
+            const Real densityTEME = earthAtmosphericModelTEME.getDensityAt(positionITRF, instant);
+
+            EXPECT_TRUE(densityITRF.isNear(densityTEME, tolerance)) << String::Format(
+                "{} ≈ {} Δ {} [T]", densityITRF.toString(), densityTEME.toString(), (densityITRF - densityTEME)
+            );
```

Comment:
I'm a bit confused. So we're comparing the ITRF density to TEME and seeing that it's within 1e-15?

Reply:
Yeah, this is a test to ensure:
1. the input Position is being converted to the correct frame before computing the density
2. to a lesser extent, that our TEME->ITRF conversion doesn't significantly affect the output density